src/view/signUpView.java

```diff
@@ -151,5 +156,8 @@ public void propertyChange(PropertyChangeEvent evt) {
         if (state.getError() != null) {
             JOptionPane.showMessageDialog(this, state.getError());
         }
+        if (state.getEmpty() != null){
+            this.instructionsText = "Please Input a Username. 🪿";
+        }
```

Comment:
Good that you have error messages for the failed view

Reply:
The signups don't have restrictions like when I only put a user name and click sign up I enter. I don't know if sign-up is completed or not, but you should add fail cases when the user enters: no password as well.

Also when I click on the signup view, initially it shows the error message "Please Input a username". I don't know if this is what you meant to code or not.